crates/server/README.md

```diff
@@ -0,0 +1,257 @@
+managing the Calimero Admin Dashboard and the P2P Applications built for Calimero Nodes.
+
+Calimero Server component is split into 3 parts:
+
+### 1. Admin API
+
+The Admin API component of the Calimero Server connects web applications with the node and its functionalities. It is primarily utilized by the Admin Dashboard to query and manage various aspects of the node, including:
+
+ - Identity information
+ - Root keys
+ - Client keys
+ - Installed applications
+ - Started contexts
+
+
+**Data Querying**: The Admin API allows the Admin Dashboard to fetch important data from the node, such as identity details, root and client keys, and information about installed and active applications.
+
+**Application Management**: The API provides functionalities to install new applications and start contexts. This enables administrators to dynamically manage the software running on the node.
+
+**Key Management**: Administrators can manage root and client keys through the API, ensuring secure access and control over the node.
+
+**Authentication**: The Admin API facilitates user authentication via selected wallets, currently supporting MetaMask and NEAR networks. Authentication details will be explained in later sections.
+
+**Integration with Web Applications**: The authentication mechanism is also used by web applications designed to interact with P2P applications installed on the node, ensuring secure and authenticated access.
+
+### 2. JSON rpc
+
+The JSON-RPC component of the Calimero Server facilitates communication between the front-end web applications of P2P systems and the application binaries (contexts) running on Calimero nodes. This allows seamless interaction and data management for applications.
+
+The JSON-RPC interface provides two primary methods:
+
+- Query
+- Mutate
+
+#### Query Method
+The `Query` method retrieves data from the applications running on the Calimero nodes. For instance, in the Only Peers forum application, posts and comments stored in the application's storage can be queried using the JSON-RPC interface. This enables users to fetch and display content from the forum.
+
+#### Mutate Method
+The `Mutate` method allows modification of the application's data. For example, in the Only Peers forum application, users can create new posts or comments. The Mutate method updates the application's storage with these new entries, facilitating dynamic content creation and interaction within the application.
+
+The JSON-RPC component is crucial for enabling interactive and real-time data management between P2P application front-ends and the back-end contexts running on Calimero nodes.
+
+### 3. Websocket
+
+The WebSocket is used for subscribing to and unsubscribing from certain contexts within the Calimero server. Defined handlers manage subscription states for WebSocket connections, allowing clients to receive updates about specific contexts they are interested in.
```

Comment:
Above JSON-RPC enables interaction with the application, and here WebSocket enable interaction with the context.